### DISCUSSION TITLE
Getting Undefined index: id error

### DISCUSSION THREAD

**Discussion by larstton** [livewire/livewire#4446] [1. Help]
I have created a livewire component:

```
`    @foreach($this->recipe->products as $key => $recipeProduct)

                                    <livewire:recipe-ingredient
                                        wire:id="{{ $key }}"
                                        :recipeProduct="$recipeProduct"
                                        :product="$recipeProduct->product">
                                    </livewire:recipe-ingredient>

       @endforeach`
```

```
<div>
     <button type="submit"
                    value="Add to Cart"
                    wire:click="addToCart()"
                    class="button h-14 w-80">
                Add to Cart
            </button>
</div>

```

When the button is clicked, I am getting an error:

Undefined index: id 

from the file: PerformActionCalls

  `      $id = $update['payload']['id'];`
  
  
  I had similar code for another component and it was working before. Livewire version 2.8.2

**Comment by larstton** [comment]
Apparently, the issue was persistent on any livewire component.  Those 2 commands fixed the issue:

php artisan livewire:publish --config

php artisan livewire:publish --assets

I have also removed the 2.8.2 version and installed 2.8.0

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by joshhanley** [comment]
@snaix93 just to confirm you managed to fix the issue right?

How come you had to downgrade back to 2.8.0 if using those commands fixed the issue?
